bin/add_metadata_to_hmm.py

```diff
@@ -0,0 +1,147 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import csv
+import gzip
+import logging
+from pathlib import Path
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def parse_ko_list(ko_list_file: Path) -> dict[str, dict[str, str]]:
+    """
+    Parses the KO list TSV file and returns a dictionary with KO ids as keys and associated data
+    (threshold, score_type, definition).
+    """
+    ko_data = {}
+    open_func = gzip.open if ko_list_file.suffix == '.gz' else open
+    
+    logging.info(f"Parsing KO list file: {ko_list_file}")
+    
+    with open_func(ko_list_file, 'rt') as f:
+        reader = csv.DictReader(f, delimiter='\t')
+        for row in reader:
+            knum = row['knum']
+            threshold = row['threshold']
+            score_type = row['score_type']
+            definition = row['definition']
+            # Only include entries that have valid 'domain' or 'full' profile type
+            if score_type in ['domain', 'full']:
+                ko_data[knum] = {
+                    'threshold': threshold,
+                    'score_type': score_type,
+                    'definition': definition
+                }
+    
+    logging.info(f"Parsed {len(ko_data)} KO entries.")
+    return ko_data
+
+
+def create_desc_line(knum: str, ko_data: dict[str, dict[str, str]]) -> str:
+    """Generate DESC line for the given KO id."""
+    return f"DESC  {ko_data[knum]['definition']}\n"
+
+
+def create_ga_line(knum: str, ko_data: dict[str, dict[str, str]]) -> str:
+    """Generate GA line based on profile type for the given KO id."""
+    threshold = ko_data[knum]['threshold']
+    score_type = ko_data[knum]['score_type']
+    
+    if score_type == 'domain':
+        return f"GA    {threshold} {threshold}\n"
+    elif score_type == 'full':
+        return f"GA    {threshold} 25.00\n"
+    else:
+        raise ValueError(f"Unknown profile type {score_type} for KO {knum}.")
+
+
+def modify_hmm_profile(hmm_file: Path, ko_data: dict[str, dict[str, str]]):
+    """
+    Modifies an HMM profile by adding DESC and GA lines and writes to a new file.
+    """
+    logging.info(f"Modifying HMM profile: {hmm_file}")
+
+    # Read the content of the HMM profile
+    with hmm_file.open('r') as f:
+        lines = f.readlines()
+
+    # Find the KO id from the 'NAME' line
+    name_line = next((line for line in lines if line.startswith('NAME')), None)
+    if name_line:
```

Comment:
Is it expected that all the hmm_files have a NAME? If so, I would make the script fail if there is any without it.

With that in mind, I would flip the code to make it more readable:
```python
if not name_line:
   raise ValueException(f"HMM {hmm_file} is missing the 'NAME' line")
```
this removes the nested code